lib/serve_static.ex

```diff
@@ -6,7 +6,7 @@ defmodule HTTPServer.ServeStatic do
     |> Map.merge(routes)
   end
 
-  defp get_filepaths(dirpath), do: Path.wildcard(dirpath <> "/*.{html,xml,txt,css,json}")
+  defp get_filepaths(dirpath), do: Path.wildcard(dirpath <> "/*.{json,html,css,xml,txt,css,jpg,gif,png}")
```

Comment:
does maintaining this list of file type extensions matter? could you just use a second wildcard, and if the file is in the directory with an extension that is not supported elsewhere, just let it 404?
```suggestion
  defp get_filepaths(dirpath), do: Path.wildcard(dirpath <> "/*.*")
```

Reply:
If it does matter, then this knowledge about supported file type exists in multiple places in this code. Here and in the HeaderBuilder.  Should it only exists in one place?

Reply:
I think I can definitely set up some error handling for this! One question: Should it be a 415 error code?

Reply:
hmmm, maybe. but if its a supported media type but the file just is not there, then it would be a 404.  ... try not to get too hung up on specifics to start. Not looking for this comment to become a feature.

Reply:
The part to focus on is having the rules live in just one place.

Reply:
Got it, that makes sense to me. I removed the supported file type boundaries from Path.wildcard and refactored `get_media_type/3` to send back a 404 if the media type isn't currently supported by the server.